tools/run-fbp-tests

```diff
@@ -44,6 +44,104 @@ import re
 import subprocess
 import sys
 
+valgrind_opts = [
+    "--show-reachable=no",
+    "--quiet",
+    "--error-exitcode=1",
+    "--leak-check=full",
+]
+
+class FbpTest:
+    SKIP_NONE = 0
+    SKIP_SIMPLE = 1
+    SKIP_VALGRIND = 2
+    SKIP_COMPILE = 3
+
+    supported_skip = {
+        "TEST-SKIP": SKIP_SIMPLE,
+        "TEST-SKIP-VALGRIND": SKIP_VALGRIND,
+        "TEST-SKIP-COMPILE": SKIP_COMPILE,
+    }
+
+    def __init__(self, test_file):
+        self.test_file = test_file
+        self.skip_msg = None
+        self.skip = self.SKIP_NONE
+        self.output = False
+        self.output_regex = False
+        self.output_spec = ""
+        self.exit_code = 0
+
+        self.__parse_test__()
+
+    def __parse_skip__(self, cmd, splitted):
+        self.skip = self.supported_skip.get(cmd, self.SKIP_NONE)
+        if not self.skip:
+            return
+
+        if len(splitted) > 2:
+            self.skip_msg = splitted[2]
+
+    def __parse_command__(self, line):
+        splitted = line.split(None, 2)
+        cmd = splitted[1]
+
+        if cmd == "TEST-OUTPUT-REGEX":
+            self.output_regex = True
+        elif cmd == "TEST-OUTPUT":
+            self.output = True
+        elif cmd == "TEST-EXPECTS-ERROR":
+            self.exit_code = 1
+        elif cmd.startswith("TEST-SKIP"):
+            self.__parse_skip__(cmd, splitted)
+        else:
+            print("Ignoring unknown test command '%s' in test %s" %
+                  (cmd, self.test_file))
+
+    def __parse_test__(self):
+        f = open(self.test_file, encoding="UTF-8")
+        for line in f:
+            if line.startswith("## "):
+                self.__parse_command__(line)
+            elif (self.output or self.output_regex) and line.startswith("#"):
+                self.output_spec += line[2:] or "\n"
+
+    def is_skip(self, skips):
+        if self.skip == self.SKIP_NONE:
+            return False
+        elif self.skip in skips:
+            return True
+        return False
+
+    def check_result(self, out, code):
```

Comment:
This is not checking the result, it's printing values to the standard output. Either rename the function or use raise an exception with the error details so the call site can print them.